### PR TITLE
Add krb5_cc_retrieve_cred() and krb5_cc_remove_cred()

### PR DESCRIPTION
**Pull request by steffen-kiess**

### CODE REVIEW

File: src/krb5/_ccache.pyx [hunk 1]
```diff
@@ -104,6 +120,38 @@ cdef extern from "python_krb5.h":
...
+    ('match_2nd_tkt', KRB5_TC_MATCH_2ND_TKT),
+    ('match_keytype', KRB5_TC_MATCH_KTYPE),
+]
+# If KRB5_TC_SUPPORTED_KTYPES is not available it will be set to 0 in
+# python_krb5.h
+if KRB5_TC_SUPPORTED_KTYPES != 0:
+    _CredentialsRetrieveFlags_members += [
+        ('supported_ktypes', KRB5_TC_SUPPORTED_KTYPES),
+    ]
+CredentialsRetrieveFlags = enum.IntEnum('CredentialsRetrieveFlags', _CredentialsRetrieveFlags_members)
```
Comment (jborean93) [inline src/krb5/_ccache.pyx:153]: Any reason why you are adding the members like this, the principal side just declared the enum like normal Python code https://github.com/jborean93/pykrb5/blob/2e5284a3076925fc8b8ab5d99ed89e8a0270986e/src/krb5/_principal.pyx#L77-L81.

**Review by jborean93 — COMMENTED** [review]
I'll have to look into the packaging side to figure out why it's failing. Any chance on adding actual tests for these new functions?

**Comment by jborean93** [comment]
https://github.com/jborean93/pykrb5/pull/24 should fix up the CI errors, it seems like Ubuntu 22.04 ships with a version of pip which is incompatible with how we test things. That PR moves the project structure back to using the `setup.cfg` style project definition which improves compatibility. If you rebase your PR it should pick up those changes for the next run.

Looking at the macOS errors it looks like you need to define the enum in the `_ccache.pyi` file for mypy to be happy. https://github.com/jborean93/pykrb5/blob/main/src/krb5/_principal.pyi#L9-L24 is how I did it for other enums.

**Comment by steffen-kiess** [comment]
> Any chance on adding actual tests for these new functions?

I can try to do that.



> That PR moves the project structure back to using the `setup.cfg` style project definition which improves compatibility. If you rebase your PR it should pick up those changes for the next run.

Ok, I'll do that.

**Comment by steffen-kiess** [comment]
I've added a test, and the CI now seems to run through.

**Review by jborean93 — APPROVED** [review]

**Comment by jborean93** [comment]
Thanks for working on this and filling out the tests and typing stuff, appreciate the work.

**Pull request merged**